django/contrib/gis/db/backends/mysql/schema.py

```diff
@@ -58,6 +58,43 @@ def remove_field(self, model, field):
 
         super().remove_field(model, field)
 
+    def _alter_field(
+        self,
+        model,
+        old_field,
+        new_field,
+        old_type,
+        new_type,
+        old_db_params,
+        new_db_params,
+        strict=False,
+    ):
+        super()._alter_field(
+            model,
+            old_field,
+            new_field,
+            old_type,
+            new_type,
+            old_db_params,
+            new_db_params,
+            strict=strict,
+        )
+
+        old_field_spatial_index = (
+            isinstance(old_field, GeometryField)
+            and old_field.spatial_index
+            and not old_field.null
+        )
+        new_field_spatial_index = (
+            isinstance(new_field, GeometryField)
+            and new_field.spatial_index
+            and not new_field.null
+        )
```

Comment:
I'm curious why this one is different to the others, when I remove the `null` checks I didn't get failures locally. Did I miss something?
```suggestion
        old_field_spatial_index = (
            isinstance(old_field, GeometryField) and old_field.spatial_index
        )
        new_field_spatial_index = (
            isinstance(new_field, GeometryField) and new_field.spatial_index
        )
```

Reply:
> Did I miss something?

Yes, MySQL doesn't support spatial indexes on `NULL` columns (20c2d625d3d5062e43918d1d7b6f623202491dd4). I can add MySQL-specific test for altering nullable field, if you think it's needed.

Reply:
Ah thank you
> I can add MySQL-specific test for altering nullable field, if you think it's needed.

If it's simple and you don't mind, then yes please, but don't worry if it's complex or awkward :+1: 
Apparently I even looked at that commit but have the memory of a goldfish.

Reply:
np, I will do this later today. I don't have my "Django" laptop with me :smile:

Reply:
Added, sorry for being so slow :sweat_smile:

Reply:
Brilliant thank you